_assets/javascripts/lib/distance-sorter.js

```diff
@@ -0,0 +1,167 @@
+    this.searchInput = this.searchForm.getElementsByTagName('input')[0];
+    this.formSubmit = this.searchForm.getElementsByTagName('button')[0];
+    this.formSubmit.disabled = false;
+    this.locationsCarousel = DistanceSorter.getLocationsCarousel();
+    this.cards = this.locationsCarousel.cards;
+    this.locationFinder = new CRDS.LocationFinder();
+  }
+
+  static getLocationsCarousel() {
+    return Object.values(CRDS._instances).find(
+      (instance) => DistanceSorter._isSectionLocationsCarousel(instance)
+    );
+  }
+
+  static _isSectionLocationsCarousel(instance) {
+    return instance.carousel !== undefined && instance.carousel.id === 'section-locations';
+  }
+
+  handleFormSubmit(event) {
+    this._disableForm();
+    this.getDistance()
+      .done((locationDistances) => { this._updateDomWithSortedCards(locationDistances); })
+      .fail((xhr, ajaxOptions, thrownError) => {
+        console.log(thrownError);
+        this.showError();
+        this.formSubmit.disabled = false;
+        this.removeLabels();
+      });
+  }
+
+  _disableForm() {
+    event.preventDefault();
+    this.formSubmit.disabled = true;
+  }
+
+  getDistance() {
+    this.locationDistances = [];
+    return this.locationFinder.getLocationDistances(this.searchInput.value);
+  }
+
+  _updateDomWithSortedCards(locationDistances) {
+    this._updateLocationDistancesArray(locationDistances);
+    this._display_sorted_locations_carousel();
+    this._resetFormOnSuccess();
+  }
+
+  _updateLocationDistancesArray(locationDistances) {
+    locationDistances.forEach(
+      locationObject => this._addToLocationDistancesArray(locationObject)
+    );
+  }
+
+  _addToLocationDistancesArray(locationObject) {
+    const locationDistance = this._createLocationDistanceObject(locationObject);
+    this.locationDistances.push(locationDistance);
+  }
+
+  _createLocationDistanceObject(locationObject) {
+    return {
+      location: locationObject.location.location,
+      distance: locationObject.distance
+    };
+  }
+
+  _display_sorted_locations_carousel() {
+    this.createDataAttributes();
+    this.appendDistances();
+    this.locationsCarousel.sortBy('distance');
+//  A anywhere card needs to be created in Contenful before
+//  this line will run correctly.
+//  this.anywhereCheck();
+  }
+
+  createDataAttributes() {
+    for (let i = 0; i < this.cards.length; i += 1) {
+      const location = this.cards[i].getElementsByClassName('card-title')[0].children[0].textContent;
+      this.cards[i].dataset.location = location;
+    }
+  }
+
+  appendDistances() {
+    for (let i = 0; i < this.cards.length; i += 1) {
+      const locationMatch = this.locationDistances.find(obj => obj.location === this.cards[i].dataset.location);
+      if (locationMatch !== undefined) {
+        this.cards[i].dataset.distance = locationMatch.distance;
+        const span = document.createElement('span');
+        span.classList.add('distance', 'label', 'font-family-base');
```

Comment:
here we can remove `font-family-base` and add `soft-half` and `push-half`. Finished product might look like this:
```
<span class="distance label soft-half push-half">4.9 miles</span>
```